cuda/components/zero_array.cu

```diff
@@ -71,6 +71,7 @@ void zero_array(size_type n, ValueType *array)
 #define GKO_DECLARE_ZERO_ARRAY(_type) \
     void zero_array<_type>(size_type n, _type * array);
 GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_ZERO_ARRAY);
+GKO_INSTANTIATE_FOR_EACH_INDEX_TYPE(GKO_DECLARE_ZERO_ARRAY);
```

Comment:
No, it's instantiating for index type instead of value types, i.e. not the same type between the two calls. AFAIK he needs to zero int arrays in another kernels, so this makes sense. Except if I missed something?

Reply:
You are correct, I did not check the name properly...
I only looked at the beginning, end and the width, I guess I was just lazy. My bad.